internal/engine/metrics/controller.go

```diff
@@ -0,0 +1,78 @@
+package metrics
+
+import (
+	"context"
+	"crypto/tls"
+
+	"contrib.go.opencensus.io/exporter/ocagent"
+	"go.opencensus.io/stats/view"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/tilt-dev/tilt/internal/store"
+	"github.com/tilt-dev/tilt/pkg/logger"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+type Controller struct {
+	exporter *DeferredExporter
+	metrics  model.MetricsSettings
+}
+
+func NewController(exporter *DeferredExporter) *Controller {
+	return &Controller{
+		exporter: exporter,
+	}
+}
+
+func (c *Controller) diff(rStore store.RStore) model.MetricsSettings {
+	state := rStore.RLockState()
+	defer rStore.RUnlockState()
+	return state.MetricsSettings
+}
+
+func (c *Controller) OnChange(ctx context.Context, rStore store.RStore) {
+	newMetrics := c.diff(rStore)
```

Comment:
this name made me think it more directly contained to actual metrics / values - how about "settings" or "metricsSettings"?